docs/src/modules/ROOT/pages/constraints-and-score/score-calculation.adoc

```diff
@@ -666,17 +1052,35 @@ collector.
 The following code snippet first groups all visits by the vehicle that will be used,
 and averages all the service durations using the `ConstraintCollectors.average(...)` collector.
 
+[tabs]
+====
+Java::
++
 [source,java,options="nowrap"]
 ----
 private Constraint averageServiceDuration(ConstraintFactory constraintFactory) {
     return constraintFactory.forEach(Visit.class)
             .groupBy(Visit::getVehicle, average(Visit::getServiceDuration))
             .penalize(HardSoftScore.ONE_SOFT,
-                    (vehicle, averageServiceDuration) -> averageServiceDuration)
+                    (vehicle, averageServiceDuration) -> (int) averageServiceDuration)
```

Comment:
Personally, I'd prefer `HardSoftLongScore` and removing the cast.
Simpler code for the example.

Reply:
`averageServiceDuration` is a double, not a long. Cast is needed regardless.